nixos/modules/services/games/asf.nix

```
@@ -0,0 +1,200 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.archisteamfarm;
+  asf-config = pkgs.writeText "ASF.json" (builtins.toJSON (cfg.config // {
+    # we disable it because ASF cannot update itself anyways
+    # and nixos takes care of restarting the service
+    # is in theory not needed as this is already the default for default builds
+    UpdateChannel = 0;
+  }));
+
+  mkBot = n: c:
+    builtins.toJSON (c.settings // {
+      SteamLogin = if c.username == "" then n else c.username;
+      SteamPassword = c.passwordFile;
+      # sets the password format to file (https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Security#file)
+      PasswordFormat = 4;
+      Enabled = c.enabled;
+      name = n;
+    });
+
+in {
+  options.services.archisteamfarm = {
+    enable = mkOption {
+      type = types.bool;
+      description = ''
+        If enabled, starts the Archis Steam Farm service.
+        For configuring the SteamGuard token you will need to use the web-ui, which is enabled by default over on 127.0.0.1:1242.
+        You cannot configure ASF in any way outside of nix, since all the config files get wiped on restart and replaced with the programatically set ones by nix.
+      '';
+      default = false;
+    };
+
+    web-ui = mkOption {
+      type = types.submodule {
+        options = {
+          enable = mkEnableOption
+            "Wheter to start the web-ui. This is the preffered way of configuring things such as the steam guard token.";
+
+          package = mkOption {
+            type = types.package;
+            default = pkgs.ASF-ui;
+            description =
+              "Web ui package to use. Contents must be in $out/lib/dist.";
+          };
+        };
+      };
+      default = {
+        enable = true;
+        package = pkgs.ASF-ui;
+      };
+      example = {
+        enable = false;
+      };
+      description = "The Web-UI hosted on 127.0.0.1:1242.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.ArchiSteamFarm;
+      description =
+        "Package to use. Should always be the latest version, for security reasons and since this module uses very new features.";
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/asf";
+      description = "State directory of Archis-Steam-Farm";
+    };
+
+    config = mkOption {
+      type = types.attrs;
```

Community consensus was to call this option `settings` instead of `config`: see https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md for details.

```suggestion
    settings = mkOption {
      type = let format = pkgs.formats.json {}; in format.type;
```